libsel4vm/include/sel4vm/guest_vm_util.h

```diff
@@ -72,6 +72,30 @@ static inline vm_vcpu_t *vm_find_free_unassigned_vcpu(vm_t *vm)
     return NULL;
 }
 
+/***
+ * @function vm_find_free_unassigned_cpu(vm)
+ * Find an unallocated physical cpu
+ * @param {vm_t *} vm           A handle to the vm owning the vcpu
+ * @return                      -1 if no pcpu can be found, otherwise the available pcpu
+ */
+static inline int vm_find_free_unassigned_cpu(vm_t *vm)
+{
+    for (int core = 0; core < vm->num_vcpus; core++) {
```

Comment:
I'm not sure I get this algorithm, shouldn't we loop over the physical cores in the outer loop and check in the innter loop if this core is used by any VM?

Reply:
Agreed, CONFIG_MAX_NUM_NODES probably makes more sense here.